docs/content/guides/dagster/local_to_prod_tutorial.mdx

```diff
@@ -0,0 +1,393 @@
+- IO manager - an IO manager is a special kind of resource that handles storing and loading assets. For example, if we wanted to store assets in S3, we would use dagster’s built in S3 IO manager.
+- run config - assets and resources sometimes require configuration to set certain values (like the password to a database). Run config allows you to set these values at run time. In this example, we will also use an API to set some default run configuration.
+
+## Problem Statement
+
+Let’s say we’ve been tasked with fetching data from Hacker News and splitting it into two datasets: one containing all of the data about stories and one containing all of the data about comments. We have the following additional requirements:
+
+- For local development, we want to store the datasets as tables in a development snowflake database called `SANDBOX` that has a schema for each user.
+- For production, we want to store the datasets as tables in a production snowflake database called `PRODUCTION` that has a schema called `HACKER_NEWS`.
+- For testing, we don’t want to fetch data from Hacker News, instead we want to provide seed data so we can assert that the data was split into the stories and comments datasets correctly.
+
+### Setup
+
+If you want to follow along with this example, you can either:
+
+Clone the example (this will give you the final version of the code):
+
+    dagster new-project --example local_to_prod
+    cd local_to_prod
+    pip install .
+
+Start with a scaffolded project and fill in the code as we go
+
+    dagster new-project --quick-start local_to_prod
+    cd local_to_prod
+    pip install dagster dagit pandas pytest dagster-snowflake dagster-snowflake-pandas
+
+### Part One: Local Development
+
+Let’s start by writing the assets
+
+```python file=/guides/dagster/local_to_prod_tutorial/assets.py startafter=start_assets endbefore=end_assets
+# assets.py
+
+@asset(
+    required_resource_keys={"hn_client"},
+)
+def items(context) -> pd.DataFrame:
+    """Items from the Hacker News API: each is a story or a comment on a story."""
+
+    hn_client = context.resources.hn_client
+
+    rows = []
+    for item_id in range(1, 100 + 1):
+        rows.append(hn_client.fetch_item_by_id(item_id))
+
+    result = pd.DataFrame(rows, columns=hn_client.item_field_names).drop_duplicates(subset=["id"])
+    result.rename(columns={"by": "user_id"}, inplace=True)
+
+    return result
+
+
+@asset
+def comments(items: pd.DataFrame) -> pd.DataFrame:
+    """Comments from the Hacker News API."""
+    return items[items["type"] == "comment"]
+
+
+@asset
+def stories(items: pd.DataFrame) -> pd.DataFrame:
+    """Stories from the Hacker News API."""
+    return items[items["type"] == "story"]
+```
+
+We’ll also need to write the resource to fetch data from Hacker News
+
+<Note>
```

Comment:
Is `<Note> `a good way to make these callout boxes? Is there another js component that would be better?

Reply:
I like it - i think `<Note>` works - we can figure out a better-looking one if necessary.